HIP 128: Onboarding fees and vendor coupon (#1056)
helium/HIP: 0128-energy-subnetwork.md

At scale, the VPP that exists via the Helium ENERGY subnetwork may be leased (purchased via Data Credits) by Service Providers and offered to Clients such as energy companies, grid operators, or aggregators in order to balance grid supply or demand (Energy Services). Use of these resources would reward individual _DER Hosts_ with ENERGY token.

#### Overview

> A typical installation consists of Solar Panels and an Inverter shown in the diagram. Optionally a Battery is added. The Inverter's responsibly is to manage the energy from the Solar Panels, Battery, Home Appliances (your normal electrical devices) and the Power Grid. The Inverter is connected to the local home network via the Internet Router (e.g. via ethernet cable or wifi). An ENERGY Gateway (e.g. a new or existing IOT Hotspot), connected to the home network via the Internet Router, allows for collection of data from the Inverter and to control its operation. The ENERGY Gateway signs the data from the Inverter so that the source of the data is known. The ENERGY Gateway can receive signed signals to control the behavior of the Inverter. Several ENERGY gateways collectively form a Virtual Power Plant where control at scale can offer ancillary services to the grid and home owners can be rewarded for this accordingly.

Reviewer: ```suggestion
A typical installation consists of solar panels and an inverter, as shown in the diagram. Optionally, one or more batteries can be added.

The inverter's responsibility is to manage the energy from the solar panels, battery, home appliances (your normal electrical devices), and the power grid. The inverter is connected to the local home network via the internet router (e.g., via Ethernet cable or Wi-Fi).

An ENERGY Gateway connected to the home network via the internet router allows for the collection of data from the inverter and control of its operation. The ENERGY Gateway signs the data from the inverter so that the source of the data is known. The ENERGY Gateway can receive signed signals to control the behavior of the inverter.
```
- apply consistent formatting
- removes repeated definitions from above

At scale, the VPP that exists via the Helium ENERGY subnetwork may be leased (purchased via Data Credits) by Service Providers and offered to Clients such as energy companies, grid operators, or aggregators in order to balance grid supply or demand (Energy Services). Use of these resources would reward individual _DER Hosts_ with ENERGY token.

#### Overview
A typical installation consists of solar panels and an inverter, as shown in the diagram. Optionally, one or more batteries can be added.

> The inverter's responsibility is to manage the energy from the solar panels, battery, home appliances (your normal electrical devices), and the power grid. The inverter is connected to the local home network via the internet router (via Ethernet cable or Wi-Fi).

Reviewer: ```suggestion
The inverter's responsibility is to manage the energy from the solar panels, battery, home appliances, and the power grid. The inverter is connected to the local home network via the internet router (via Ethernet cable or Wi-Fi).
```

#### Overview
A typical installation consists of Solar Panels and an Inverter shown in the diagram. Optionally a Battery is added. The Inverter's responsibly is to manage the energy from the Solar Panels, Battery, Home Appliances (your normal electrical devices) and the Power Grid. The Inverter is connected to the local home network via the Internet Router (e.g. via ethernet cable or wifi). An ENERGY Gateway (e.g. a new or existing IOT Hotspot), connected to the home network via the Internet Router, allows for collection of data from the Inverter and to control its operation. The ENERGY Gateway signs the data from the Inverter so that the source of the data is known. The ENERGY Gateway can receive signed signals to control the behavior of the Inverter. Several ENERGY gateways collectively form a Virtual Power Plant where control at scale can offer ancillary services to the grid and home owners can be rewarded for this accordingly.

```mermaid

> %%{init: {'theme': 'neutral'}}%%
> graph LR
> 
>     Internet[Internet] -.- InternetRouter
>     Meter<---->|AC| PowerGrid[Power Grid]
> 
>     subgraph Home [Home]
>         Meter-->|AC|HomeApp
>         Meter[Utility Meter]<-->|AC|Inverter
> 
>         Solar[Solar Panels\nDER] -->|DC| Inverter[Inverter]
>         style Solar stroke-width: 3px
>         Battery[Battery Storage\nDER] <-->|DC| Inverter
>         style Battery stroke-width: 3px
>         Inverter -->|AC| HomeApp[Home Appliances]
> 
>         style Gateway stroke-dasharray: 5,5
>         InternetRouter[Gateway Controller] -.- Inverter
>         InternetRouter[Internet Router] -.- Gateway[ENERGY Gateway\n#40;IOT Hotspot#41;]
>     end

Reviewer: I think this serves as some light cleanup here.
```suggestion
%%{init: {'theme': 'neutral'}}%%
graph LR
    Internet[Internet] -.- InternetRouter
    Meter<---->|AC| PowerGrid[Power Grid]
    subgraph Home [Home]
        Meter-->|AC|HomeApp
        Meter[Utility Meter]<-->|AC|Inverter
        Solar[Solar Panels\nDER] ==>|DC| Inverter[Inverter]
        Battery[Battery Storage\nDER] <==>|DC| Inverter
        Inverter -->|AC| HomeApp[Home Appliances]
        InternetRouter[Gateway Controller] -.- Inverter
        InternetRouter[Internet Router] -.- Gateway[ENERGY Gateway\n#40;IOT Hotspot#41;]
        style Gateway stroke-width: 3
    end
```

```mermaid
%%{init: {'theme': 'neutral'}}%%
graph LR
    Internet[Internet] -.- InternetRouter
    Meter<---->|AC| PowerGrid[Power Grid]
    subgraph Home [Home]
        Meter-->|AC|HomeApp
        Meter[Utility Meter]<-->|AC|Inverter
        Solar[Solar Panels\nDER] ==>|DC| Inverter[Inverter]
        Battery[Battery Storage\nDER] <==>|DC| Inverter
        Inverter -->|AC| HomeApp[Home Appliances]
        InternetRouter[Gateway Controller] -.- Inverter
        InternetRouter[Internet Router] -.- Gateway[ENERGY Gateway\n#40;IOT Hotspot#41;]
        style Gateway stroke-width: 3
    end
```

Gateways can manage multiple DERs on a local network. Under a gateway, each DER is onboarded individually. In many cases this will be just one inverter for a household, but as the network evolves, additional entities may be onboarded – such as pool heaters, batteries, or even bitcoin miners.

The fee for onboarding an ENERGY gateway is set at $10 USD, paid using DC and the fee for onboarding one DER is set at $20 USD, paid using the ENERGY token.

> Gateway vendors can issue coupons to hotspots for the onboarding fees. If a coupon is redeemed and the vendor do not have enough DC/ENERGY to cover it, twice the amount will be slashed (50% will go to an insurance fund and 50% will be burned) from the vendor funds. When a vendor is below the minimum stake after slashing they are not allowed to issue coupons.

Reviewer: ```suggestion
Gateway vendors can issue coupons for the onboarding of gateways and their respective fees. If a coupon is redeemed and the vendor does not have enough DC/ENERGY to pay from their onboarding account, twice the amount will be slashed from the vendor's stake. Of the slashed amount, 50% will go to an insurance fund and 50% will be burned. When a vendor's stake is depleted, they are not allowed to issue onboarding coupons.
```

> The location of the entity is defined as part of the gateway onboarding. There is no separate location assertion fee. DERs are linked to the gateway by an on-chain mechanism.

Reviewer: ```suggestion
The location of the entity is defined as part of the DER onboarding. There is no separate location assertion fee.
```

Applying the governance structure of the Helium community to power systems will increase the numbers of those involved in the influence and buildout of yet another very critical part of our infrastructure. Communication is essential for both sensors and people, but electrical energy is at the core of a modern society. Now is the time to be part of the start in the shift to a new model that is operated by the people - for the people.

### Note on Current Onboarding Fee

> We have implemented a 10 USD in DC onboarding fee due to the current structure of the A variable in the DAO utility score. For future subnetworks, we believe that the main proxy for the utility score should be DC burn for services, rather than this onboarding fee. We propose that the A score should be reconsidered in favor of using DC service burn as the proxy for network value, as the current onboarding fee may not be optimal for new subnetworks joining on fair terms.

Reviewer 2: ```suggestion
We have implemented a 10 USD in DC onboarding fee due to the current structure of the A variable in the DAO utility score. For future subnetworks, we believe that the main proxy for the utility score should be DC burn for services, rather than this onboarding fee. Srcful believes that the A score should be reconsidered in favor of using DC service burn as the proxy for network value, as the current onboarding fee may not be optimal for new subnetworks joining on fair terms.
```
might be good to clarify this.

Applying the governance structure of the Helium community to power systems will increase the numbers of those involved in the influence and buildout of yet another very critical part of our infrastructure. Communication is essential for both sensors and people, but electrical energy is at the core of a modern society. Now is the time to be part of the start in the shift to a new model that is operated by the people - for the people.

### Note on Current Onboarding Fee

> We have implemented a 10 USD in DC onboarding fee due to the current structure of the A variable in the DAO utility score. For future subnetworks, we believe that the main proxy for the utility score should be DC burn for services, rather than this onboarding fee. Srcful believes that the A score should be reconsidered in favor of using DC service burn as the proxy for network value, as the current onboarding fee may not be optimal for new subnetworks joining on fair terms.

Reviewer: ```suggestion
We have implemented a $10 USD onboarding fee in DC due to the current structure of the A variable in the DAO utility score. For future subnetworks, we believe that the main proxy for the utility score should be DC burn for services rather than this onboarding fee. Srcful believes that the A score should be reconsidered in favor of using DC service burn as the proxy for network value, as the current onboarding fee may not be optimal for new subnetworks joining on fair terms.
```
